CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+# Minimum required CMake version
+cmake_minimum_required(VERSION 3.15)
+
+# Project name
+project(llama_cpp)
+
+# Git repository location
+set(REPO_URL "https://github.com/ggerganov/llama.cpp")
+
+# Requirements file
+set(REQUIREMENTS_FILE "requirements.txt")
+
+# Llama directory
+set(LLAMA_DIR "../src/llama_cpp")
+
+# Check for Python and Git using CMake's FIND_PACKAGE
+find_package(PythonLibs REQUIRED)
+find_package(Git REQUIRED)
+
```

Comment:
Make sure, these two commands are not required with cmake

touch $(LLAMA_DIR)/__init__.py
 cp $(LLAMA_DIR)/convert-hf-to-gguf.py $(LLAMA_DIR)/convert_hf_to_gguf.py